src/services/redisSessionStorage.ts

```diff
@@ -432,8 +423,52 @@ export default class RedisSessionStorage extends RedisStorageUtils
             _ !== `${sessionInfoKeyPrefix}${user.session_token}`
         )
         .map(_ => `${sessionKeyPrefix}${_.split(sessionInfoKeyPrefix)[1]}`);
+      /**
+       * Retrieve all user wallet tokens related to session tokens.
+       * Wallet tokens are stored inside user payload.
+       */
+      const errorOrUserWalletTokens = await new Promise<
+        Either<Error, ReadonlyArray<string>>
+      >(resolve => {
+        this.redisClient.mget(...sessionKeys, (err, response) =>
+          resolve(this.arrayStringReply(err, response))
+        );
+      });
+      // Map every user payload with an Option<WalletToken>
+      // If the value is invalid or must be skipped, it will be mapped with none
+      const walletsToken = errorOrUserWalletTokens
+        .fold(
+          _ => [],
+          _ =>
+            _.map(value => {
+              try {
```

Comment:
is there some methods inside this try / catch that may throw ? (it looks like no method can actually throw). is fo, we can omit this try / catch part.

Reply:
Fixed in 60f3781
I've also removed code duplication.